### PR TITLE
feat: Add cli for listing all module accounts

### PR DESCRIPTION
**Pull request by mattverse**
cref: https://github.com/cosmos/cosmos-sdk/pull/9812

## Description

This PR adds a CLI for querying all module accounts, `module-accounts`, in the auth module. 
Using this command would display all the account information, including human readable name of the module, module account address, account number, permissions, etc. 

This command would be especially useful for developers using cosmos-sdk as there are lots of instances where a developer would have to look into the module account, but only to do so by getting the module account address after getting the module name, hashing it and bech32fying it in the status quo. By using this command, users would be able to get a quick look on the list of all module accounts' information.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [x] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [x] reviewed tests and test coverage
- [x] manually tested (if applicable)

### CODE REVIEW

**Review by ValarDragon — APPROVED** [review]
im assuming this is a copy of the upstream PR

File: x/auth/keeper/grpc_query.go [hunk 1]
```diff
@@ -83,3 +83,28 @@ func (ak AccountKeeper) Params(c context.Context, req *types.QueryParamsRequest)
...
+	for moduleName := range ak.permAddrs {
+		account := ak.GetModuleAccount(ctx, moduleName)
+		if account == nil {
+			return nil, status.Errorf(codes.NotFound, "account %s not found", moduleName)
+		}
+		any, err := codectypes.NewAnyWithValue(account)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, err.Error())
+		}
+		modAccounts = append(modAccounts, any)
```
Comment (alexanderbez) [inline x/auth/keeper/grpc_query.go:106]: Since this is pre-allocated, I would use direct indexing instead of appending.

**Review by alexanderbez — APPROVED** [review]
LGTM! Just one minor nit :p

**Comment by mattverse** [comment]
Yeah this is a cherry pick from the upstream

File: x/auth/keeper/grpc_query.go (hunk 1 above)
Comment (mattverse) [reply x/auth/keeper/grpc_query.go:106]: 👍

Comment (ValarDragon) [reply x/auth/keeper/grpc_query.go:106]: Since this is from upstream, lets leave as-is?

**Comment by mattverse** [comment]
reverting the commit, merging it identical to upstream

**Review by alexanderbez — APPROVED** [review]

File: x/auth/keeper/grpc_query.go (hunk 1 above)
Comment (alexanderbez) [reply x/auth/keeper/grpc_query.go:106]: Okiee :p

**Pull request merged**